### PR TITLE
IGVF-1111-samples-donors-dependency

### PR DESCRIPTION
**Pull request by ian-whaling**

### CODE REVIEW

File: src/igvfd/types/file_set.py [hunk 1]
```diff
@@ -288,6 +310,29 @@ def summary(self, request, assay_term, preferred_assay_title=None, readout=None,
                 sentence += phrase
         return sentence
 
+    @calculated_property(
+        condition='samples',
+        schema={
+            'title': 'Donors',
+            'description': 'The donors of the samples associated with this measurement set.',
```
Comment (gabdank) [inline src/igvfd/types/file_set.py:317]: curated sets rather than measurment

File: src/igvfd/types/file_set.py [hunk 2]
```diff
@@ -385,6 +430,29 @@ def summary(self, request, file_set_type, measurement_sets=None):
...
+            'type': 'array',
+            'uniqueItems': True,
+            'items': {
+                'title': 'Donor',
+                'description': 'Donor of a sample associated with this measurement set.',
+                'type': 'string',
```
Comment (gabdank) [inline src/igvfd/types/file_set.py:443]: auxiliary sets rather than measurement

File: src/igvfd/types/file_set.py [hunk 3]
```diff
@@ -104,6 +103,29 @@ def assay_title(self, request, input_file_sets=None):
...
+                'linkTo': 'Donor'
```
Comment (gabdank) [inline src/igvfd/types/file_set.py:117]: analysis sets rather than measurement?

**Review by gabdank — CHANGES_REQUESTED** [review]
I think comments in the donor calculation were copied form measurment set, and were supposed to be class specific. I also think that the function itself is duplicated 3 times - so maybe we should refactor and have it called 3 times, rather than have duplicated code

File: src/igvfd/types/file_set.py (hunk 1 above)
Comment (ian-whaling) [reply src/igvfd/types/file_set.py:317]: this is for measurement sets. curated sets do not calculate donors, the property is still submittable for them.

File: src/igvfd/types/file_set.py (hunk 2 above)
Comment (ian-whaling) [reply src/igvfd/types/file_set.py:443]: fixed

File: src/igvfd/types/file_set.py (hunk 3 above)
Comment (ian-whaling) [reply src/igvfd/types/file_set.py:117]: fixed

File: src/igvfd/schemas/prediction_set.json [hunk 4]
```diff
@@ -4,10 +4,23 @@
     "$id": "/profiles/prediction_set.json",
     "$schema": "https://json-schema.org/draft/2020-12/schema",
     "type": "object",
-    "required": [
-        "lab",
-        "award",
-        "file_set_type"
+    "anyOf": [
```
Comment (gabdank) [inline src/igvfd/schemas/prediction_set.json:7]: I think it should be oneOf

**Review by gabdank — CHANGES_REQUESTED** [review]
I thin it should be oneOf and not anyOf

**Review by gabdank — APPROVED** [review]

**Pull request closed without merging**